### PR TITLE
Remove Ruby proof-of-concept code

### PR DESCRIPTION
**Pull request by 0xdaryl**
An integration of OMR into the YARV Ruby VM was used as a proof-of-concept
demonstration of the versatility and relative simplicity of integrating
the technology into another language environment.
    
The project never matured beyond the proof-of-concept stage though, and
many of the parts that surfaced in OMR have gone stale and cannot be tested
or maintained.

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@rwy0717 or @charliegracie , please review the GC change.

File: gc/startup/omrgcalloc.cpp [hunk 1]
```diff
@@ -33,16 +33,6 @@
...
-	MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(omrVMThread);
-
-	/* TODO: Deprecate this -- required for early versions of ruby/omr integration */
-	if (allocator->isGCAllowed() && (NULL == env->getExtensions()->getGlobalCollector())) {
-		/* Lazily create the collector before attempting to allocate. */
-		if (OMR_ERROR_NONE != OMR_GC_InitializeCollector(omrVMThread)) {
-			return NULL;
-		}
-	}
-
```
Comment (rwy0717) [inline gc/startup/omrgcalloc.cpp]: LGTM

Comment (charliegracie) [reply gc/startup/omrgcalloc.cpp]: Can we assert that the global collector is NOT null.

**Comment by 0xdaryl** [comment]
Marking this as a WIP while I address comments and to let #3557 merge first.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
Rebased, added a GC assert, and un-WIPed.

**Comment by 0xdaryl** [comment]
Can a committer merge this please?

**Comment by fjeremic** [comment]
Upps disregard that @Leonardo2718 I noticed the circle looking icon in the Reviewers section on the side bar so I clicked it to see what it does. Not it's apparent. It asks for a re-review.

**Pull request merged**